packages/server/graphql/types/Notification.ts

```diff
@@ -39,7 +39,7 @@ export const notificationInterfaceFields = {
   }
 }
 
-const Notification = new GraphQLInterfaceType({
+const Notification: GraphQLInterfaceType = new GraphQLInterfaceType({
```

Comment:
+1 hint, the place where TS got lost is the `resolveTypeLookup` object. If you do the following
```
    const resolveTypeLookup: Record<string, GraphQLObjectType<any, GQLContext>> = {
      KICKED_OUT: NotifyKickedOut,
      PAYMENT_REJECTED: NotifyPaymentRejected,
      TASK_INVOLVES: NotifyTaskInvolves,
      PROMOTE_TO_BILLING_LEADER: NotifyPromoteToOrgLeader,
      TEAM_ARCHIVED: NotifyTeamArchived,
      TEAM_INVITATION: NotificationTeamInvitation,
      MEETING_STAGE_TIME_LIMIT_END: NotificationMeetingStageTimeLimitEnd
    } as const
```
TS will figure out the type of `Notification` automatically

Reply:
It looks like `const Notification: GraphQLInterfaceType` or `const resolveTypeLookup: Record<string, GraphQLObjectType<any, GQLContext>> = {` fixes the no implicit any error. As `const Notification: GraphQLInterfaceType` is slightly more concise, I'll stick with that for now. Same applies with the few comments below. 

While in the neighbourhood, I started adding resolve types. I usually did this by adding `resolveType: ({type}: {type: keyof typeof resolveTypeLookup}) => {` I'm open to better ideas as I don't love that `type` isn't necessarily a key of the resolveTypeLookup. Alternatively, I could create a union type of the different resolve type keys, but we'd need to maintain that too.